src/main/java/com/artipie/asto/s3/S3Storage.java

```diff
@@ -108,7 +109,13 @@ public S3Storage(final S3AsyncClient client, final String bucket) {
 
     @Override
     public CompletableFuture<Void> move(final Key source, final Key destination) {
-        throw new UnsupportedOperationException();
+        return this.client.copyObject(
```

Comment:
@olegmoz The `source` object from storage should be removed, since we are moving it to a new location.

Reply:
@paulodamaso very good point, thank you! Added original object deletion now and added test for it.